lib/auth/machineid/machineidv1/bot_service.go

```diff
@@ -541,36 +545,26 @@ func (bs *BotService) deleteBotRole(ctx context.Context, botName string) error {
 // deleteBotAuthz allows the legacy rbac noun/verbs to continue being used until
 // v16.0.0.
 func (bs *BotService) deleteBotAuthz(ctx context.Context) error {
-	var authCtx *authz.Context
-	var originalErr error
-	authCtx, originalErr = authz.AuthorizeWithVerbs(
-		ctx, bs.logger, bs.authorizer, false, types.KindBot, types.VerbDelete,
-	)
-	if originalErr != nil {
+	authCtx, err := authz.Authorize(ctx, bs.authorizer, bs.logger)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	if authErr := authCtx.CheckAccessToKind(ctx, false, types.KindBot, types.VerbDelete); authErr != nil {
 		// TODO(noah): DELETE IN 16.0.0
-		var err error
-		authCtx, err = authz.AuthorizeWithVerbs(
-			ctx, bs.logger, bs.authorizer, false, types.KindUser, types.VerbDelete,
-		)
-		if err != nil {
-			return originalErr
+		if err := authCtx.CheckAccessToKind(ctx, false, types.KindUser, types.VerbDelete); err != nil {
+			return authErr
```

Comment:
nit:

```suggestion
			return trace.Wrap(authErr)
```

Same for following lines.